src/instruments/analysis.zig

```diff
@@ -0,0 +1,55 @@
+const std = @import("std");
+const runner_fifo = @import("../runner_fifo.zig");
+const shared = @import("../shared.zig");
+
+pub const AnalysisInstrument = struct {
+    fifo: runner_fifo.RunnerFifo,
+
+    const Self = @This();
+
+    pub fn init(allocator: std.mem.Allocator) !Self {
```

Comment:
Why don't we share most of this file's logic with the perf instrument ? Since the only non common part of this is the init where we ask the mode, it should be incorporated in the shared struct, and just take a mode as argument of initialisation. The rest of this file could easily be dropped without losing anything.

Each instrument is then just a small wrapper around the init call with the proper argument.
Keep in mind I'm not familiar with zig tho